src/main/java/edu/cornell/kfs/sys/service/impl/DocumentMaintenanceServiceImpl.java

```diff
@@ -1,45 +1,100 @@
 
+import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Iterator;
+import java.util.List;
 
 public class DocumentMaintenanceServiceImpl implements DocumentMaintenanceService {
     private static final Logger LOG = LogManager.getLogger(DocumentMaintenanceServiceImpl.class);
-
-    public static final int WAIT_TIME_NO_WAIT = 0;
-
     private DocumentMaintenanceDao documentMaintenanceDao;
 
     @Transactional
     public boolean requeueDocuments() {
         Collection<String> docIds = documentMaintenanceDao.getDocumentRequeueValues();
         LOG.info("requeueDocuments: Total number of documents flagged for requeuing: " + docIds.size());
 
+        List<ActionItemNoteDetailDto> noteDetails = documentMaintenanceDao.getActionNotesToBeRequeued();
+        LOG.info("requeueDocuments: Total number of action note details: " + noteDetails.size());
+        
         for (String docId : docIds) {
             LOG.info("requeueDocuments: Requesting requeue for document: " + docId);
-            DocumentRefreshQueue documentRequeuer = KewApiServiceLocator.getDocumentRequeuerService(CoreConfigHelper.getApplicationId(), docId, WAIT_TIME_NO_WAIT);
+            DocumentRefreshQueue documentRequeuer = (DocumentRefreshQueue) SpringContext.getService(CUKFSConstants.DOCUMENT_REFRESH_QUEUE_SERVICE_SPRING_CONTEXT_NAME);
             documentRequeuer.refreshDocument(docId);
+            
+            List<ActionItemNoteDetailDto> noteDetailsForDocument = findNoteDetailsForDocument(noteDetails, docId);
+            for (ActionItemNoteDetailDto detailDto : noteDetailsForDocument) {
+                List<ActionItem> actionItems = findActionItem(detailDto);
+                if (CollectionUtils.isNotEmpty(actionItems)) {
+                    ActionItem item = actionItems.get(0);
+                    if (ObjectUtils.isNotNull(item)) {
+                        ActionItemExtension actionItemExtension = findActionItemEtencsion(item.getId());
+                        if (ObjectUtils.isNull(actionItemExtension)) {
+                            actionItemExtension = buildActionItemExtension(detailDto, item);
+                            LOG.info("requeueDocuments, adding note details " + detailDto.toString() + " to action item " + item.getId());
+                        } else {
+                            actionItemExtension.setActionNote(detailDto.getActionNote());
+                            actionItemExtension.setNoteTimeStamp(detailDto.getNoteTimeStamp());
+                            LOG.info("requeueDocuments, updating note details " + detailDto.toString() + " to action item " + item.getId());
+                        }
+                        KRADServiceLocator.getDataObjectService().save(actionItemExtension);
+                    }
+                }
+            }
         }
-
         return true;
     }
+    
+    private ActionItemExtension findActionItemEtencsion(String actionItemId) {

```

Comment:
Minor typo in method name, should be changed from "findActionItemEtencsion" to "findActionItemExtension"

Reply:
OK, I will fix